### PR TITLE
Adding Week 7 Blogpost

### PR DESCRIPTION
**Pull request by ganimtron-10**
Adding Weekly Blogpost.
msedge_yrubCVqDSN

### CODE REVIEW

File: docs/source/posts/2022/2022-07-20-week-7-praneeth.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- Storing, managing, and transforming is difficult.
+
+.. image:: https://user-images.githubusercontent.com/64432063/181731181-1f242c65-ccb8-4589-a2ed-40bfb3718cfd.gif
+    :width: 400
+    :align: center
+
+
+Did you get stuck anywhere?
+---------------------------
+It was hard to debug why the tests were failing in ``ubuntu`` and ``mac os``. I tried investing it by installing ubuntu and got nothing but then while implementing the new selection approach it automatically got fixed.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:47]: Missing comma after `got nothing`

Comment (m-agour) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:47]: Maybe `Ubuntu` and `macOS`

File: docs/source/posts/2022/2022-07-20-week-7-praneeth.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+.. image:: https://user-images.githubusercontent.com/64432063/181730428-debd0617-dc32-4232-93ab-18ab903e92de.gif
+    :width: 400
+    :align: center
+
+I also tried to implement a freehand drawing mode by adding ``Disk2D`` as points. But as you can see below, there are many flaws in this.
+
+- First of all, we add many ``Disk2D`` objects which make it memory consuming process.
+- If the mouse moves very fast then we can see the gaps between two points.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:37]: Missing comma after `very fast`

Comment (m-agour) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:37]: Maybe just `between points`

File: docs/source/posts/2022/2022-07-20-week-7-praneeth.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+What did you do this week?
+--------------------------
+I continued PR `#623`_ and fixed the displacement of the shape from its original position when applying rotation. This was happening because all the results were type casted to ``int`` due to which some positions got rounded off. As we call this function continuously, each time the round-off would happen, the shape would get displaced.
+
+.. image:: https://user-images.githubusercontent.com/64432063/181723334-ef9ec75d-d3bf-4b79-83bf-272545c4dd12.gif
+    :width: 400
+    :align: center
+
+For fixing this I converted all the calculations into integer calculations using floor division ``//`` instead of normal division ``/``.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:19]: Missing comma after `For fixing this`

File: docs/source/posts/2022/2022-07-20-week-7-praneeth.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+.. image:: https://user-images.githubusercontent.com/64432063/181730428-debd0617-dc32-4232-93ab-18ab903e92de.gif
+    :width: 400
+    :align: center
+
+I also tried to implement a freehand drawing mode by adding ``Disk2D`` as points. But as you can see below, there are many flaws in this.
+
+- First of all, we add many ``Disk2D`` objects which make it memory consuming process.
+- If the mouse moves very fast then we can see the gaps between two points.
+- Storing, managing, and transforming is difficult.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:38]: Should be `are difficult`

File: docs/source/posts/2022/2022-07-20-week-7-praneeth.rst [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+.. post:: July 20 2022
+   :author: Praneeth Shetty 
+   :tags: google
+   :category: gsoc
+
+
+What did you do this week?
+--------------------------
+I continued PR `#623`_ and fixed the displacement of the shape from its original position when applying rotation. This was happening because all the results were type casted to ``int`` due to which some positions got rounded off. As we call this function continuously, each time the round-off would happen, the shape would get displaced.
```
Comment (m-agour) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:13]: Missing a comma after `casted to int`. 
Should be `cast`. I'd also suggest rephrasing the sentence for more clarity.

**Review by m-agour — APPROVED** [review]
Hello @ganimtron-10,
Please see my comments below. Thanks.

File: docs/source/posts/2022/2022-07-20-week-7-praneeth.rst [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    :width: 400
+    :align: center
+
+For fixing this I converted all the calculations into integer calculations using floor division ``//`` instead of normal division ``/``.
+
+.. image:: https://user-images.githubusercontent.com/64432063/181723783-352092aa-b7a7-4d13-8d26-553315c7e1aa.gif
+    :width: 400
+    :align: center
+
+The tests were failing in the `ubuntu` and the `mac os` because the mouse click event wasn't propagated to the line due to some unknown reasons. When investigating more about the issue, Mohamed suggested and helped me implement another approach to select the shapes. In this approach, we calculate the distance between the mouse click event position and each of the shapes, and if any of the distances is less than a specific value (which I call as limit value) then we send this current event to that element.
```
Comment (xtanion) [inline docs/source/posts/2022/2022-07-20-week-7-praneeth.rst:25]: missing comma after `(which I call as limit value)`

**Review by xtanion — COMMENTED** [review]
Hi @ganimtron-10, Really nice and detailed blog. There are a few punctuation mistakes as I mentioned below, PTAL.
Thanks.

**Comment by ganimtron-10** [comment]
Thank You @m-agour and @xtanion for the review!

**Review by skoudoro — APPROVED** [review]
LGTM ! Thanks @ganimtron-10, merging

**Pull request merged**